data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -24,15 +42,96 @@
-    LOG.info("Initiated the start() method...");
+    sourceConfig.getTopics().forEach(topic -> {
+      totalWorkers = 0;
+      Properties consumerProperties = getConsumerProperties(topic);
+      try {
+        totalWorkers = topic.getTopic().getConsumerGroupConfig().getWorkers();
+        consumerGroup = topic.getTopic().getConsumerGroupConfig().getGroupName();
+        executorService = Executors.newFixedThreadPool(totalWorkers);
+        IntStream.range(0, totalWorkers).forEach(index -> {
+          String consumerId = consumerGroup +"::"+ topic.getTopic().getName()+"::"+Integer.toString(index + 1);
+          multithreadedConsumer = new MultithreadedConsumer(consumerId,
+                  consumerGroup, consumerProperties, topic.getTopic(), buffer, pluginMetrics);
+          consumers.add(multithreadedConsumer);
+          executorService.submit(multithreadedConsumer);
+        });
+      } catch (Exception e) {
+        LOG.error("Failed to setup the Kafka Source Plugin.", e);
+        kafkaWorkerThreadProcessingErrors.increment();
+      }
+    });
   }
 
   @Override
   public void stop() {
-   LOG.info("Initiated the stop method...");
+    LOG.info("Shutting down Consumers...");
+    executorService.shutdown();
+    try {
+      if (!executorService.awaitTermination(
+              sourceConfig.getTopics().get(0).getTopic().getConsumerGroupConfig().getThreadWaitingTime().toSeconds(), TimeUnit.MILLISECONDS)) {
+        LOG.info("Consumer threads are waiting for shutting down...");
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      if (e.getCause() instanceof InterruptedException) {
+        LOG.error("Interrupted during consumer shutdown, exiting uncleanly...");
+        executorService.shutdownNow();
+        Thread.currentThread().interrupt();
+      }
+    }
+    LOG.info("Consumer shutdown successfully...");
+  }
+
+  private Properties getConsumerProperties(TopicsConfig topicConfig) {
+    Properties properties = new Properties();
+    try {
+      properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommitInterval().toSecondsPart());
+      properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoOffsetReset());
+      properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
+      properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommit());
+      properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getTopic().getConsumerGroupConfig().getGroupId());
+      setPropertiesForSchemaType(topicConfig, properties, topicConfig.getTopic().getSchemaConfig().getSchemaType());
+    } catch (Exception e) {
+      LOG.error("Unable to create the Kafka Consumer from the given configurations.", e);
```

Comment:
will this re-throw the exception? because I guess we would want to fail kafka source initialization..